### ISSUE TITLE
CI: SQLite adapter is running in Trilogy tests

### ISSUE THREAD

**Issue by skipkayhil** [rails/rails#50579]
Ref #49373

https://buildkite.com/rails/rails/builds/103367#018cd514-4e51-4bac-be7c-c8656888c21e

I'm pretty sure this means that some other test is leaking, but I have not dug in at all.

**Comment by rafaelfranca** [comment]
Yes. That is a leak. The connections aren't properly removed after they are established. We need to fix https://github.com/rails/rails/pull/49382
